### PR TITLE
HPCC-24946 Add jlog multiline logger methods

### PR DESCRIPTION
**Pull request by jakesmith**
Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by jakesmith** [comment]
@richardkchapman - please review.
NB: I did not remove/replace the multiLog version that's in thorsoapcall, because it's also being used to translate HTTP requests and processing carriage returns (replacing them with spaces),

File: common/thorhelper/thorsoapcall.cpp [hunk 1]
```diff
@@ -1645,7 +1645,7 @@ class CWSCAsyncFor : implements IWSCAsyncFor, public CInterface, public CAsyncFo
         content = contentDecoded;
 
         if (soapTraceLevel > 6 || master->logXML)
-            multiLog(master->logctx, "Content decoded. Original %s %d", CONTENT_LENGTH, contentLength);
+            master->logctx.CTXLOG("Content decoded. Original " CONTENT_LENGTH " %d", contentLength);
```
Comment (richardkchapman) [inline common/thorhelper/thorsoapcall.cpp:1648]: Is this supposed to say mCTXLOG ?

Comment (jakesmith) [reply common/thorhelper/thorsoapcall.cpp:1648]: no, deliberately changed, as can't be multi line

Comment (richardkchapman) [reply common/thorhelper/thorsoapcall.cpp:1648]: Not sure about this change. I think you've decided Not to log the contents any more - but there seems to be an explicit setting master->logXML requesting that you do...

Comment (jakesmith) [reply common/thorhelper/thorsoapcall.cpp:1648]: It wasn't/isn't logging the content just a static string,CONTENT_LENGTH :
```
#define CONTENT_LENGTH "Content-Length: "
```

**Comment by richardkchapman** [comment]
Not sure about the changes in thorsoapcall - replacing \r actually seems like it would be valuable for the other multilog cases on occasion, and I don't understand the comment about translating HTTP...

Other than that looks ok.

**Comment by jakesmith** [comment]
@richardkchapman - please see change to handle carriage return and replace multiLog in thorsoapcall

**Comment by richardkchapman** [comment]
@jakesmith Better - please squash

**Comment by jakesmith** [comment]
@richardkchapman - squashed

**Pull request merged**